File: 0152-maximum-product-subarray/0152-maximum-product-subarray.java
class Solution {
    public int maxProduct(int[] nums) {
        int m=Integer.MIN_VALUE;
        int p=1,s=1;
        for(int i=0;i<nums.length;i++){
            if(p==0) p=1;
            if(s==0) s=1;
            p*=nums[i];
            s*=nums[nums.length-i-1];
            m=Math.max(m,Math.max(p,s));
        }
        return m;
    }
}

File: 0152-maximum-product-subarray/0152-maximum-product-subarray.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int s=INT_MIN;
        int pre=1,suf=1;
        for(int i=0;i<nums.size();i++){
            if(pre==0) pre=1; 
            if(suf==0) suf=1;
            pre=pre*nums[i];
            suf=suf*nums[nums.size()-i-1];
            s=max(s,max(pre,suf));}
        return s;
    }
};